lib/redux/reducers/users.js

```diff
@@ -91,14 +103,22 @@ export function targetDevices(state = {}, action) {
     }
     // create some scaffolding based on the users the loggedInUser
     // currently has upload access to
-    // (since what's in localStorage could be outdated)
     case actionTypes.LOGIN_SUCCESS:
     case actionTypes.SET_USER_INFO_FROM_TOKEN: {
       const { memberships } = action.payload;
       let newState = {};
       _.each(memberships, (membership) => {
         if (isPwd(membership)) {
-          newState[membership.userid] = [];
+          const membershipTargetDevices = _.get(membership, ['profile', 'patient', 'targetDevices'], []);
```

Comment:
Couldn't this just be:
`newState[membership.userid] = membershipTargetDevices`? I don't think all the `if`/`else`-ing is necessary.